tests/test_base.cpp

```diff
@@ -43,23 +43,30 @@ static const char *ioserv_path()
 	return result;
 }
 
-void test_wrapper_with_session::operator() () const
+void configure_session(newapi::session &session, const session_create_args &session_args)
 {
-	dnet_node *n;
 	std::vector<int> groups;
 	uint64_t cflags;
 	uint64_t ioflags;
-	std::tie(n, groups, cflags, ioflags) = session_args;
-
-	newapi::session session(n);
-	auto log = session.get_logger();
-	DNET_LOG_INFO(log, "Start test: {}", test_name);
+	std::tie(std::ignore, groups, cflags, ioflags) = session_args;
 
 	session.set_groups(groups);
 	session.set_cflags(cflags);
 	session.set_ioflags(ioflags);
 
 	session.set_exceptions_policy(session::no_exceptions);
+}
+
+void test_wrapper_with_session::operator() () const
```

Comment:
Let's keep `test_wrapper_with_session` as it was and setup `session` directly instead of adding and using `configure_session()`.

Reply:
I extracted some staff to `configure_session()` function, because it is used in tests in this PR. In source branch user hasn't any choice, user is forced to configure both test and session in one place. Not flexible. User hasn't any stuff to configure session in test body, or to configure more than one session per test.

I can make `configure_session()` as copy-paste, If you want me not to touch old `test_wrapper_with_session::operator()`.